Makefile

```diff
@@ -569,18 +571,28 @@ $(LANTERN_MOBILE_ANDROID_SDK): $(ANDROID_SDK)
 	mkdir -p $(LANTERN_MOBILE_LIBS) && \
 	cp $(ANDROID_SDK) $(LANTERN_MOBILE_ANDROID_SDK)
 
-$(LANTERN_MOBILE): $(LANTERN_MOBILE_TUN2SOCKS) $(LANTERN_MOBILE_ANDROID_LIB) $(LANTERN_MOBILE_ANDROID_SDK)
+android-dependencies: $(LANTERN_MOBILE_TUN2SOCKS) $(LANTERN_MOBILE_ANDROID_LIB) $(LANTERN_MOBILE_ANDROID_SDK)
 	cd $(LANTERN_MOBILE_DIR)/app
-	gradle -b $(LANTERN_MOBILE_DIR)/app/build.gradle \
+	cp $(ANDROID_SDK_ANDROID_LIB) $(LANTERN_MOBILE_LIBS)
+
+android-debug: android-dependencies
+	gradle -PlanternVersion=$(GIT_REVISION) -b $(LANTERN_MOBILE_DIR)/app/build.gradle \
 		clean \
 		assembleDebug
 
-android: $(LANTERN_MOBILE)
```

Comment:
The reason that I used $(LANTERN_MOBILE) is that naming targets after the output file that they create allows Make to be smart about not rebuilding them if the output file already exists.  Setting it up as you have it, we will always rebuild the app whenever we run a dependent target like install.  You can fix this by defining targets for $(LANTERN_MOBILE_ANDROID_DEBUG) and $(LANTERN_MOBILE_ANDROID_RELEASE) and then making android-debug and android-release depend on those.